controllers/front/OrderController.php

```diff
@@ -237,6 +237,45 @@ public function displayAjaxselectDeliveryOption()
         ]));
     }
 
+    public function displayAjaxCheckCustomerInformation(): void
+    {
+        $email = Tools::getValue('email');
+        $isGuestAllowed = (bool) Configuration::get('PS_GUEST_CHECKOUT_ENABLED');
+        $customerExists = false;
+        $responseData = [];
+        $alert = [];
+
+        if (Validate::isEmail($email)) {
+            $customerExists = Customer::customerExists($email, false, true);
+        }
+
+        if ($customerExists) {
+            $alert['type'] = $isGuestAllowed ? 'info' : 'danger';
+            $alert['message'] = $isGuestAllowed ?
+                $this->translator->trans(
+                    'You already have an account, don\'t hesitate to sign in!',
+                    [],
+                    'Shop.Theme.Checkout'
+                )
+                :
+                $this->translator->trans(
+                    'An account was already registered with this email address',
+                    [],
+                    'Shop.Notifications.Error'
+                );
+        }
+
+        $responseData = array_merge($responseData, [
```

Comment:
```suggestion
        $responseData = [
```

$responseData is never filled.